functions-python/feed_sync_process_transitland/src/main.py

```diff
@@ -73,404 +37,156 @@ def log_message(level, message):
-            )
-            raise
-
-    def check_feed_url_exists(self, feed_url: str) -> bool:
-        """
-        Check if a feed with the given URL exists in any state (active or deprecated).
-        This check is used to prevent creating new feeds with URLs that are already in use.
-
-        Args:
-            feed_url (str): The URL to check
-
-        Returns:
-            bool: True if any feed with this URL exists (either active or deprecated),
-                 preventing creation of new feeds with duplicate URLs
-        """
-        results = self.session.query(Feed).filter(Feed.producer_url == feed_url).all()
-
-        if results:
-            if len(results) > 1:
-                log_message("warning", f"Multiple feeds found with URL: {feed_url}")
-                return True
-
-            result = results[0]
-            if result.status == "active":
-                log_message(
-                    "info", f"Found existing feed with URL: {feed_url} (status: active)"
-                )
-                return True
-            elif result.status == "deprecated":
-                log_message(
-                    "error",
-                    f"Feed URL {feed_url} exists in deprecated feed (id: {result.id}). "
-                    "Cannot reuse URLs from deprecated feeds.",
-                )
-                return True
-
-        log_message("debug", f"No existing feed found with URL: {feed_url}")
-        return False
-
-    def get_current_feed_info(
-        self, external_id: str, source: str
-    ) -> Tuple[Optional[str], Optional[str]]:
-        """
-        Get current feed ID and URL for given external ID
-
-        Args:
-            external_id (str): The external ID to look up
-            source (str): The source of the feed
-
-        Returns:
-            Tuple[Optional[str], Optional[str]]: Tuple of (feed_id, feed_url)
-        """
-        result = (
+    def _get_current_feeds(self, external_id: str, source: str) -> List[Feed]:
+        """Retrieve current feeds for a given external ID and source."""
+        return (
             self.session.query(Feed)
             .filter(Feed.externalids.any(associated_id=external_id, source=source))
-            .first()
+            .all()
         )
-        if result is not None:
-            log_message(
-                "info",
-                f"Retrieved feed {result.stable_id} "
-                f"info for external_id: {external_id} (status: {result.status})",
-            )
-            return result.id, result.producer_url
-        log_message("info", f"No existing feed found for external_id: {external_id}")
-        return None, None
-
-    def publish_to_batch_topic(self, payload: FeedPayload) -> None:
-        """
-        Publish feed to dataset batch topic
 
-        Args:
-            payload (FeedPayload): The feed payload to publish
-        """
+    def _update_feed(self, payload: FeedPayload, old_feed_id: Optional[str]) -> Feed:
```

Comment:
Rename this function to clarify that it is `deprecating` a feed in favor of a new feed.